tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```diff
@@ -143,6 +143,21 @@ spec:
           local identity=$1
           local reference=$2
           local digest=$3
+
+          # cosign has very limited support for selecting the right auth entry,
+          # so create a custom auth file with just one entry.
+          SOURCE_AUTH_FILE_DIR=$(mktemp -d)
+          mkdir -p "${SOURCE_AUTH_FILE_DIR}"  # the tests mock `mktemp`, so we ensure it exists
+          SOURCE_AUTH_FILE="${SOURCE_AUTH_FILE_DIR}/config.json"
+
+          registry=$(echo "$reference" | cut -d '/' -f 1)
+          source_repo=${reference%%@sha256:*}
+          select-oci-auth "${source_repo}" | jq -c \
+          '.auths."'"$source_repo"'" = .auths."'"$registry"'"' > "$SOURCE_AUTH_FILE"
```

Comment:
I would suggest using `jq --arg`, but why do you need to modify it? Isn't that what `select-oci-auth` returns?

Look at my test of this here: https://github.com/konflux-ci/oras-container/pull/172/files#diff-22aaa9e3295958632657acd547ee5ee621da42d1e9fb0a6d81dc4f0f7d2986e8R170

When you run `select-oci-auth quay.io/myorg/myrepo` it will find the `quay.io/myorg/myrepo` entry in your config, but the json that is returned will say just `quay.io`.

Reply:
thanks. after more testing, I see that I can simplify the `select-oci-auth` call. Please review.